src/main/java/hillel/spring/doctors/DoctorService.java

```diff
@@ -49,39 +58,54 @@ public Doctor createDoctor(Doctor doctor) {
     }
 
     public void update(Doctor doctor) {
-        if (doctorRepository.existsById(doctor.getId()))
+        log.debug(LOG, "updating doctor with id: " +doctor.getId());
+
+        if (doctorRepository.existsById(doctor.getId())) {
             doctorRepository.save(doctor);
-        else
+        } else {
             throw new NoSuchDoctorException();
+        }
+        log.debug(LOG, "updated doctor with id: " +doctor.getId());
+
     }
 
     public boolean delete(Integer id) {
+
+        log.debug(LOG, "deleting doctor with id: "+ id);
+
         if (doctorRepository.existsById(id)) {
             doctorRepository.deleteById(id);
+            log.debug(LOG, "deleted doctor with id: "+ id);
             return true;
         }
+        log.debug(LOG, "not deleted doctor with id: "+ id);
         return false;
     }
 
 
-    public Page<Doctor> findAll(Optional<String> letter, Optional <List<String>> specialization, Pageable pageable) {
+    public Page<Doctor> findAll(Optional<String> letter, Optional<List<String>> specialization, Pageable pageable) {
         if (specialization.isPresent() && letter.isPresent()) {
+            log.debug(LOG, " doctor is found with letter and specialization: "+ letter +" " + specialization);
```

Comment:
`log.debug("doctor is found with letter and specialization: {} {} ", letter, specialization);`